build_scripts/ubuntu_sim_nuttx.sh

```diff
@@ -22,7 +22,7 @@ sudo apt-get install python-serial openocd \
     libftdi-dev libtool zlib1g-dev -y
 
 # Clean up old GCC
-sudo apt-get remove gcc-arm-none-eabi gdb-arm-none-eabi binutils-arm-none-eabi gcc-arm-embedded -y
+sudo apt-get remove gcc-arm-none-eabi gdb-arm-none-eabi binutils-arm-none-eabi gcc-arm-embedded software-properties-common -y
```

Comment:
How come this script need this but ubuntu.py does not?

Reply:
It's required for `add-apt-repository` which is not used in ubuntu.sh.

Reply:
Yeah, but they are both setting up the same environment right - so the question is really "why does one need this and not the other". 

I'm hoping to dump this script as soon as we're happy ubuntu.sh does the job adequately.